wagtail/admin/filters.py

```diff
@@ -87,7 +87,38 @@ class FilteredModelChoiceFilter(django_filters.ModelChoiceFilter):
     field_class = FilteredModelChoiceField
 
 
+class LocaleFilter(django_filters.ChoiceFilter):
+    def filter(self, qs, language_code):
+        if language_code:
+            return qs.filter(locale__language_code=language_code)
+        return qs
+
+
 class WagtailFilterSet(django_filters.FilterSet):
+    @classmethod
+    def get_filters(cls):
```

Comment:
Alternatively could break this into separate overrides of `get_filter_name`, `filter_for_field`, etc. but I opted to override the final `get_filters` instead to keep things in one place.